### PR TITLE
feat(flags): add javascript onboarding

### PR DESCRIPTION
**Pull request by michellewzhang**
adds javascript onboarding to the feature flag sidebar. also, if we detect that `event.contexts.flags` is not undefined, we  hide the first half of the onboarding as specified in the designs

### updated copy 
SCR-20241121-koun

### no integration yet 

SCR-20241120-mhqa
SCR-20241120-mhrb

https://github.com/user-attachments/assets/3fd3eade-aac4-4635-9fc9-ab5a71886651

### integration detected

https://github.com/user-attachments/assets/7c42a145-2a03-40ab-b066-279edeeed140

SCR-20241120-mrni

- relates to https://github.com/getsentry/team-replay/issues/502?reload=1?reload=1

### todo: 
- [x] analytics charts
- [ ] figure out CTA details (another way the sidebar can be triggered)

### CODE REVIEW

File: static/app/components/events/featureFlags/featureFlagOnboardingSidebar.tsx [hunk 1]
```diff
@@ -146,6 +150,10 @@ function OnboardingContent({
   hasDocs: boolean;
 }) {
   const organization = useOrganization();
+  const ORIGINAL_HASH = useMemo(() => {
+    return window.location.hash;
+  }, []);
```
Comment (billyvg) [inline static/app/components/events/featureFlags/featureFlagOnboardingSidebar.tsx:155]: Why does this need a `useMemo`?

File: static/app/components/events/featureFlags/useFeatureFlagOnboarding.tsx [hunk 2]
```diff
@@ -21,9 +24,20 @@ export function useFeatureFlagOnboarding() {
...
-    window.location.hash = 'flag-sidequest';
+    window.location.hash = FLAG_HASH;
     SidebarPanelStore.activatePanel(SidebarPanelKey.FEATURE_FLAG_ONBOARDING);
   }, []);
 
-  return {activateSidebar};
+  // if we detect that event.contexts.flags is set, use this hook instead
+  // to skip the configure step
+  const activateSidebarSkipConfigure = useCallback(
+    (event: {preventDefault: () => void}) => {
```
Comment (billyvg) [inline static/app/components/events/featureFlags/useFeatureFlagOnboarding.tsx:34]: Can we use a more specific type here?

I think `React.MouseEvent`?

File: static/app/components/events/featureFlags/utils.tsx [hunk 3]
```diff
@@ -115,31 +115,14 @@ export const sortedFlags = ({
...
-    pythonModule: 'launchdarkly',
-    pythonIntegration: 'LaunchDarklyIntegration',
-  },
-  [ProviderOptions.OPENFEATURE]: {
-    pythonModule: 'OpenFeature',
-    pythonIntegration: 'OpenFeatureIntegration',
-  },
+export const PROVIDER_OPTION_TO_URLS: Record<ProviderOptions, string> = {
+  [ProviderOptions.LAUNCHDARKLY]:
+    'https://app.launchdarkly.com/settings/integrations/webhooks/new?q=Webhooks',
```
Comment (billyvg) [inline static/app/components/events/featureFlags/utils.tsx:127]: what happened to open feature?

File: static/app/components/onboarding/gettingStartedDoc/utils/useCurrentProjectState.tsx [hunk 4]
```diff
@@ -39,7 +39,6 @@ function useCurrentProjectState({
 
   useEffect(() => {
     if (!isActive) {
-      setCurrentProject(undefined);
```
[the remark this answers is not in the document]
Comment (michellewzhang) [reply static/app/components/onboarding/gettingStartedDoc/utils/useCurrentProjectState.tsx:42]: yes - i forgot to publish my comment earlier, sorry: https://github.com/getsentry/sentry/pull/81078#discussion_r1851091411

File: static/app/components/events/featureFlags/utils.tsx (hunk 3 above)
Comment (michellewzhang) [reply static/app/components/events/featureFlags/utils.tsx:127]: so i had to separate out the two terms: **integration** and **provider**. in my onboarding code, **integration** refers to either open feature or whatever sdk is selected in the second dropdown. so if OF is selected, then the integration is OF. the **provider** would be launchdarkly (the second dropdown). 

SCR-20241121-kkpf

if the second radio button is clicked, then the **integration** and **provider** are both LD. 

when we add more options to the OF second dropdown (say "custom"), then we'll add to the provider list. but those aren't integrations ("custom" would not be an integration)

File: static/app/components/events/featureFlags/featureFlagOnboardingSidebar.tsx (hunk 1 above)
Comment (billyvg) [reply static/app/components/events/featureFlags/featureFlagOnboardingSidebar.tsx:155]: Can we leave a comment? that's weird it changes to unset -- are we changing the hash in this component?

File: static/app/components/onboarding/gettingStartedDoc/utils/useCurrentProjectState.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (billyvg) [reply static/app/components/onboarding/gettingStartedDoc/utils/useCurrentProjectState.tsx:42]: hmm do we know why it was there in the first place? are there other components that use this hook that is maybe expecting this behavior?

**Review by billyvg — APPROVED** [review]

File: static/app/components/events/featureFlags/featureFlagOnboardingSidebar.tsx (hunk 1 above)
Comment (michellewzhang) [reply static/app/components/events/featureFlags/featureFlagOnboardingSidebar.tsx:155]: we aren't changing it at all ... i couldn't figure out why it was happening😕

File: static/app/components/onboarding/gettingStartedDoc/utils/useCurrentProjectState.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (michellewzhang) [reply static/app/components/onboarding/gettingStartedDoc/utils/useCurrentProjectState.tsx:42]: hmm good point. i'll remove it for now

**Pull request merged**